docs/docs/dev_docs/contracts/compiling.md

```diff
@@ -1,22 +1,136 @@
 # Compiling contracts
 
-Please use the [TUTORIAL-TEMPLATE](../../TUTORIAL_TEMPLATE.md) for standalone guides / tutorials.
+Once you have written your [contracts](../contracts/main.md) in Noir, you will need to compile them into an artifact in order to use them. In this guide we will cover how to do so, both using the CLI and programmatically, as well as generating helper typescript and Noir interfaces for interacting with them.
 
-:::danger TODO
-TODO: this entire page
+## Prerequisites
+
+You will need the Noir build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:
+
+```bash
+curl -L https://raw.githubusercontent.com/noir-lang/noirup/main/install | bash
```

Comment:
Is it best to execute this from the home (`~`) directory?

Reply:
No need, you should be able to run it from wherever you want